source/translated_table_model.ts

```diff
@@ -28,22 +45,158 @@ export class TranslatedTableModel extends TableModel {
    *                      to this table's columnCount.
    * @throws RangeError - The index specified is not within range.
    */
-  public moveRow(source: number, destination: number): void {}
+  public moveRow(source: number, destination: number): void {
+    if(source >= this.rowCount || source < 0) {
+      throw RangeError();
+    }
+    if(destination >= this.rowCount || destination < 0) {
+      throw RangeError();
+    }
+    this.beginTransaction();
+    const traveller = (() => {
+      if(this.references[source] >= 0) {
+        return this.references[source];
+      } else {
+        return source;
+      }
+    })();
+    if(source > destination) {
```

Comment:
What if ```source == destination```?

Reply:
Should it throw a error if source is the same as destination?

Reply:
Anytime you have a question, I will bug you to ask your question as well as propose an answer to it.

Reply:
Right now if source===destination it is treated like any other move.

I feel like the correct behaviour  is that it returns without any effect rather than throwing. (And that is what was done in ArrayTableModel so that would be consistent.

Reply:
It's great when someone answers their own question. Do it.